discord/ext/commands/core.py

```diff
@@ -1094,13 +1094,14 @@ def signature(self):
             # for typing.Literal[...], typing.Optional[typing.Literal[...]], and Greedy[typing.Literal[...]], the
             # parameter signature is a literal list of it's values
             annotation = param.annotation.converter if greedy else param.annotation
-            origin = get_typing_origin(annotation)
+            origin = getattr(annotation, '__origin__', None)
             if not greedy and origin is Union:
-                union_args = get_typing_args(annotation)
-                optional = union_args[-1] is type(None)
+                none_cls = type(None)
+                union_args = annotation.__args__
+                optional = none_cls in union_args
                 if optional:
-                    annotation = union_args[0]
-                    origin = get_typing_origin(annotation)
+                    annotation = union_args[not union_args.index(none_cls)]
```

Comment:
This is kind of unreadable. Not to mention `index` raises `ValueError` if not found.